qiskit/visualization/dag_visualization.py

```diff
@@ -73,7 +74,17 @@
 
 @_optionals.HAS_GRAPHVIZ.require_in_call
 @_optionals.HAS_PIL.require_in_call
-def dag_drawer(dag, scale=0.7, filename=None, style="color"):
+def dag_drawer(
+    dag,
+    scale=0.7,
+    filename=None,
+    style="color",
+    node_attr_fn: Union[
+        Callable[[Union[DAGOpNode, DAGInNode, DAGOutNode]], dict[str, str]], None
+    ] = None,
+    edge_attr_fn: Union[Callable[[Union[Qubit, Clbit]], dict[str, str]], None] = None,
+    graph_attr: Union[dict[str, str], None] = None,
```

Comment:
I'm wondering if we want to keep the interface this low level? Right now it's completely free form and this lets you override everything `dag_drawer` will do out of the box including changing node labels. But I was thinking for this issue having a bit more prescriptive of an API that allows customizing the colors or styles might be a bit better so people don't need to learn how graphviz works. One option would be something like
 ```
 input_node_color: str,
 output_node_color: str,
 op_node_color: str,
 ```
 and then document what are valid choices for each string. The other is to take a page from the matplotlib circuit drawer and use style sheets see: https://github.com/Qiskit/qiskit/blob/492005e67c8153189352ef6ef8f9fae8b5b0c407/qiskit/visualization/circuit/qcstyle.py#L106-L151 where you take an input object that defines the style customization options for the visualization and lets users build something more domain specific for customizing the output of the visualization.

Reply:
I can understand why that may be an issue. It would certainly allow people to customize the graphs more easily by giving a simpler API. However, by doing that, you lose the ability to color nodes based on the type of the node, like in the example I provided. If that's not an issue, I can absolutely simplify the arguments or use stylesheets.

I could also use style sheets to restrict the **return value** of the input functions to `node_attr_fn`, `edge_attr_fn` and inputs to `graph_attr` to a certain subset of inputs from a defined style sheet. This would keep the flexibility, but add enough support documentation to prevent users from having to look elsewhere for the options needed to customize the DAG.

What do you think seems like a good approach?

Reply:
I sat on this for a while and changed my mind - I am in the process of implementing the stylesheet method. I like how it is similar to the matplotlib circuit drawer, and we can easily just create unique options for coloring certain nodes and edges differently.